src/io_utils.rs

```diff
@@ -0,0 +1,119 @@
+use crate::payment_store::{PaymentInfo, PAYMENT_INFO_PERSISTENCE_PREFIX};
+use crate::{Config, FilesystemLogger, NetworkGraph, Scorer};
+
+use lightning::routing::scoring::{ProbabilisticScorer, ProbabilisticScoringParameters};
+use lightning::util::ser::{Readable, ReadableArgs};
+use lightning_persister::FilesystemPersister;
+
+use rand::{thread_rng, RngCore};
+
+use std::fs;
+use std::io::{BufReader, Write};
+use std::os::unix::io::AsRawFd;
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+pub(crate) fn read_or_generate_seed_file(config: &Config) -> [u8; 32] {
```

Comment:
more of a question about how bindings work, but does file IO translate through bindings to the platform-specific APIs (`FileManager` on iOS, `SharedStorage` on Android)? Or will users be expected to implement these on their own?